doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -1517,6 +1517,9 @@ the following attributes names are recognized:
     .. warn:: Tactic Notation @qualid is deprecated since @string. @string.
        :undocumented:
 
+    .. warn:: Notation @string is deprecated since @string. @string.
+       :undocumented:
+
```

Comment:
It could be more explicit, telling which `@string` is which between `since` and `note`. It is probably worth also to give the syntax for the version number (use of dash, of dots, prefxed with a `V` or not, etc.).

Reply:
I wouldn't constrain the syntax for the version number given that if people use it for their own libraries, it could be "is deprecated since My Awesome Library 0.9" or whatever they want to.

Here is a suggestion for the more explicit distinction:

```rst
.. warn:: Notation @string__1 is deprecated since @string__2. @string__3

   :n:`@string__1` is the actual notation, :n:`@string__2` is the version number,
   :n:`@string__3` is the note.
```

Reply:
The suggestion for `string` is good. For the version number, the ability to use it in a library is indeed relevant. I'm still afraid though that we introduce inconsistency within the code that we manage, if we don't write a rule somewhere telling how it should be. Remember also that there is an option `-compat X.Y` (parsed in `G_vernac.parse_compat_version`) and that this version is legitimatedly supposed to have an action on the activation of the deprecation flag. So, maybe can we have two syntaxes for the version number, one unconstrained for libraries, and one constrained so as to work in agreement with the `-compat` option?

Reply:
I deliberately changed the semantics of compatibility notations, as I think they made little sense. Tying it to Coq versions is very strange, as is their activation depending on `-compat`. I think the warning-based approach is more sensible, especially for library authors.

The only part I kept (and I'm not fully sure about) is that deprecated notations become parsing only, so that the printer does not pick up deprecated notations. @ggonthier do you think that's the way it should be?

Reply:
I fully support removing the link between the `-compat` option and deprecated compatibility notations. This is what I was complaining about in #6840 and that's why I'm saying this PR fixes this bug.

Reply:
I'm a bit lost. Do I understand correctly that the complaint in #6840 was about the line `if is_active_compat se.synext_compat` in `metasyntax.ml`, i.e. about the semantics change induced by `-compat`? What I was talking about above was instead about the activation of the deprecation warning. Wouldn't that be a bit strange if you use `-compat 8.9` and get a warning that it is deprecated since 8.10? Anyway, that's not a big deal, and it can be amended in further PRs, so do your best.

About deactivating the printing when deprecated, why not. By the way, what do you intend to do about printing when the warning is silenced?

Reply:
> The only part I kept (and I'm not fully sure about) is that deprecated notations become parsing only, so that the printer does not pick up deprecated notations. @ggonthier do you think that's the way it should be?

I think so, although this is only meaningful for definition, since one rarely views proof terms.
One feature I think would be useful, perhaps even more so than the `since` argument, would be the ability to specify a warning category, so that the warnings related to a particular library, and perhaps a particular update of a library, could be easily selected.

Reply:
> Wouldn't that be a bit strange if you use `-compat 8.9` and get a warning that it is deprecated since 8.10?

I don't think so. The `-compat` mechanism should be kept as simple as possible IMHO and be limited to the loading of the appropriate `Compat.v` file.

Reply:
> The -compat mechanism should be kept as simple as possible IMHO and be limited to the loading of the appropriate `Compat.v` file.

Ok, that's defensible.

Reply:
Suggestion applied, thanks.